.github/workflows/release.yml

```diff
@@ -159,14 +169,31 @@ jobs:
         source ./sourceme.sh && optic_build_for_release
     - run: echo "//registry.npmjs.org/:_authToken=${{ secrets.PUBLISH_NPM_OPTIC_BOT }}" > ~/.npmrc
     - run: node ./workspaces/scripts/publish.js
+      if: false
       env:
         OPTIC_PUBLISH_SCOPE: public
+    - name: Set is_prerelease output
+      id: release_type
+      run: |
+        OPTIC_VERSION=$(node -e "console.log(require('./workspaces/local-cli/package.json').version)")
+        if [[ $OPTIC_VERSION == *'-'* ]]; then
```

Comment:
so this is determining if this is a prerelease based on whether or not there's a hyphen in the version? i feel like that could be written to be more clear with a regex instead of globbing... maybe,

```
bash-3.2$ [[ "8.5.0" =~ (.+-.+) ]]; echo $?
[[ "8.5.0" =~ (.+-.+) ]]; echo $?
1

bash-3.2$ [[ "8.5.0-blah" =~ (.+-.+) ]]; echo $?
[[ "8.5.0-blah" =~ (.+-.+) ]]; echo $?
0
```

having written that and looking at it—meh, im cool with whatever. although i do wish there was more explicit check to make, like all prerelease versions contained "beta" or "prerelease" or something

Reply:
> having written that and looking at it—meh, im cool with whatever. although i do wish there was more explicit check to make, like all prerelease versions contained "beta" or "prerelease" or something

I went through the exact same thing, thought I'd just keep it simple. We're writing versions through a library that only returns valid semver. Anything that isn't semver would have failed the publish commands outright. The prerelease id (alpha, beta, rc, etc) is supposed to be a free string in semver, but is also allowed to be omitted, so adding an optional match for that in a regex doesn't actually yield any additional safety.

Reply:
yeah, im cool with that 👍